mpp/shmem.h.in

```diff
@@ -18,6 +18,10 @@
 
 #include <stddef.h>
 #include <complex.h>
+#ifdef __cplusplus
+#define complex _Complex
```

Comment:
Interesting question. I'd be open to making that change, but I'm worried about leaving `complex` defined as a macro when `shmem.h` is included in C++ code, because of the name-shadowing problem. If `complex.h` still defines the `complex` macro when included in C++ (and I haven't seen anything forbidding it), this will break a C++ user attempting to use `std::complex`. In terms of unexpected behavior, `#undef`ing a macro seems less bad than name-masking a standard library type in possibly build-breaking ways.